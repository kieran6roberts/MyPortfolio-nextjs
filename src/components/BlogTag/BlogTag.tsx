export default function BlogTag({ tagName }: { tagName: string }) {
    return (
        <span className="px-3 py-1 mx-4 my-2 border border-purple-200 text-xxs">
            -{tagName}-
        </span>
    );
}