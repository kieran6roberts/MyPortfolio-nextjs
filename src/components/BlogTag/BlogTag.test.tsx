import * as React from "react";
import TestRenderer from "react-test-renderer";
import { render, RenderResult } from "@testing-library/react";

import BlogTag from "@/components/BlogTag/BlogTag";

let documentBody: RenderResult;

describe("<BlogTag {...props} />", () => {

    test("renders", () => {
        documentBody = render(<BlogTag  tagName="beginner" />);
        
        expect(documentBody.getByText(/beginner/i)).toBeInTheDocument();
    });

    test("renders fallback option", () => {
        documentBody = render(<BlogTag />);
        
        expect(documentBody.getByText(/unable to retrieve tag name/i)).toBeInTheDocument();
    });

    test("render snapshot", () => {
        const tree = TestRenderer
        .create(<BlogTag tagName="beginner" />)
        .toJSON();

        expect(tree).toMatchSnapshot();
    });
});