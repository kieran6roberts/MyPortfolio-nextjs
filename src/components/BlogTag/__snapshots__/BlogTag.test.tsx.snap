// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogTag {...props} /> render snapshot 1`] = `
<span
  className="px-3 py-1 mx-4 my-2 border border-purple-200 text-xxs"
>
  -
  beginner
  -
</span>
`;
